File: TP_webApp/TP_webApp/Model/Lifeline.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TP_webApp.Model
{
    public class Lifeline
    {
        

        public Lifeline(String name, String type, List<OccurenceSpecification> occurenceSpecifications)
        {
            this.name = name;
            this.type = type;
            this.occurenceSpecifications = occurenceSpecifications;
        }

        public String name { get; set; }
        public String type { get; set; }
        public List<OccurenceSpecification> occurenceSpecifications { get; set; }
    }
}

File: TP_webApp/TP_webApp/app/model/Lifeline.ts
import { OccurenceSpecification } from "./OccurenceSpecification";
import { LayerElement } from "./LayerElement";

export class Lifeline extends LayerElement {

    protected _name: string;
    protected __occurenceSpecifications: OccurenceSpecification[] = [];
    
    public get name(): string {
        return this._name;
    }

    public set name(name: string) {
        this._name = name;
    }

    public get occurenceSpecifications(): OccurenceSpecification[] {
        return this.__occurenceSpecifications;
    }

    public set occurenceSpecifications(occurenceSpecifications: OccurenceSpecification[]) {
        this.__occurenceSpecifications = occurenceSpecifications;
    }
}